src/common/Page.ts

```diff
@@ -1675,6 +1675,11 @@ export class Page extends EventEmitter {
     });
     let clip = options.clip ? processClip(options.clip) : undefined;
 
+    // Not hiding the size overlay will cause unwilling artefacts on the screenshot.
+    await this._client.send('Overlay.setShowViewportSizeOnResize', {
```

Comment:
Probably we need to set it back to the original value afterwards? Btw, have you found why the viewport is being resized when you take a screenshot?

Reply:
> Probably we need to set it back to the original value afterwards?

I initially thought this too, but then couldn't immediately think of any reason a Puppeteer script would need the overlays to render. Are you thinking of headful scenarios where the Puppeteer scripts does some set up on the page and then hands it over to the user for some manual interaction?

Reply:
Hm, I am not sure. What is the use case for running puppeteer with devtools: true? can you manipulate DevTools window as well? A hypothetical scenario of a test: resizing a window & evaluating that viewport overlay is rendered. Also, for DevTools it'd be probably helpful to capture screenshots with overlays in e2e tests at some point.

Reply:
DevTools doesn't set it back, and there is no method to get the initial value.

Reply:
I see. So do you know why the page is resized when taking a screenshot?

Reply:
> Hm, I am not sure. What is the use case for running puppeteer with devtools: true? 

We've build a complete test runner around puppeteer over at https://github.com/boxine/pentf/ . To make test inspection as easy as possible for our developers we have a `-d, --debug` flag that opens up the browser in headful mode with devtools already opened, with persistent logging enabled and a bunch of other stuff. They love it because it saves them quite a few clicks.

I'm using the same runner to test the devtools extensions for Preact (frontend framework) https://github.com/preactjs/preact-devtools/ . Currently I'm faking the extension inside an iframe, but the long term goal is to test the extension pages inside the devtools panel directly. This would properly test it like the user would experience it and would allow me to verify things like "is my extension tab showing up?", syncing extension settings, etc.